src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -46,11 +46,9 @@ private[xml] object TypeCast {
   private[xml] def castTo(
       datum: String,
       castType: DataType,
-      options: XmlOptions,
-      nullable: Boolean = true): Any = {
-    if (datum == options.nullValue &&
-      nullable ||
-      (options.treatEmptyValuesAsNulls && datum == "")){
+      options: XmlOptions): Any = {
+    if ((datum == options.nullValue) ||
```

Comment:
`nullable` was not used - always `true`. The code path here doesn't actually care about nullability of the column, which is ... OK ? if it returns `null` where it's not allowed, because the input said it was null, that's an error later anyway and probably correctly.